### ISSUE TITLE
Java - java.lang.UnsatisfiedLinkError: tensorflow_jni.dll: Can't find dependent libraries (only GPU version)

### ISSUE THREAD

**Issue by leomelki** [tensorflow/tensorflow#23656]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): Tried to compile it from source and tried binary file
- TensorFlow version: 1.12.0
- Python version: 3.7
- Bazel version (if compiling from source): 15.0
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA 10.0 & cuDNN 7.3.0
- GPU model and memory: GTX 970 TI

**Describe the problem**
Hello, I'm actually trying to contact you here because I'm stuck on that error for 2 days and it is frustrating.
When I'm trying to use tensorflow_jni.dll for GPU, it says that error but when I use the CPU version, it works. The problem is that by using CPU, it take 15 seconds to recognize one image using Yolo... with my i7 4770k... I installed Visual Studio 2017, 2015, Visual Studio Build Tool 2015 and Visual C++ 2015 Redistributable. I am using java 8

**Source code / logs**
It does that error when tensorflow_jni.dll is being loaded by java.
Exception in thread "main" java.lang.UnsatisfiedLinkError: C:\Programmation\test\NeuralNetwork\src\org\tensorflow\native\tensorflow_jni.dll: Can't find dependent libraries

Thank you really much and have a good day !

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wt-huang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by saudet** [comment]
@leomao Try to use Dependencies on tensorflow_jni.dll to see which DLL exactly is missing:
https://github.com/lucasg/Dependencies

**Comment by ymodak** [comment]
I see that you are using python 3.7. Note that python 3.7 will be supported in TF 1.13 official version.
However the good news is that TF 1.13.0-rc0 comes pre-built cuda 10 binaries.

**Comment by ymodak** [comment]
TF 1.13.1 has released and it supports Python 3.7. Please give it a try. Thanks!

**Comment by pacher** [comment]
Hello, I am having the same problem as topic starter
@saudet I don't see any suspicious in DependenciesGui. Any tips where to look and how to use it? Thanks!
@ymodak Is python required to run tf java? It is not mentioned anywhere on https://www.tensorflow.org/install/lang_java... If I install anaconda and tf for python, will it magically fix missing dependencies?
Thanks in advance

**Comment by carlosuc3m** [comment]
Hi there. Is there any update on this issue only for Java?

**Comment by saudet** [comment]
This most likely doesn't occur with JavaCPP so please give that a try: https://github.com/tensorflow/java
If it's still not working for some reason, please open an new issue over there. Thanks!

**Comment by carlosuc3m** [comment]
@saudet Sorry I am not very used to work with native libraries. Does that mean that I have to create my own wrapper of the tensorflow native classes i want to use and load them using JavaCPP?
Thank you for your quick answer

**Comment by saudet** [comment]
No, the wrappers from this repository are no longer being maintaining.
Development has moved to this repository: https://github.com/tensorflow/java

**Comment by carlosuc3m** [comment]
Sorry but I am not getting the point. What should be the approach then?

**Comment by saudet** [comment]
Try to use the artifacts made available from that subproject.

**Comment by carlosuc3m** [comment]
Thank you. However the maven artifacts of that subproject are not even ready yet, are they?

**Comment by saudet** [comment]
No, they're not. We're waiting on @ewilderj for build machines.
The default ones from GitHub Actions have an issue: https://github.com/tensorflow/java/pull/25

/cc @karllessard

**Comment by carlosuc3m** [comment]
Thank you very much. 
And just in case someone finds it useful, the problem that was causing this error was that I was not using the correct CUDA version. I tried for tf 1.12 and tf 1.15. For tf 1.12 I needed CUDA 9.0 and for tf 1.15 I needed CUDA 10.0
